src/serialization/serialization.rs

```diff
@@ -0,0 +1,412 @@
+extern crate byteorder;
+
+use super::{Counter, Histogram};
+use std::io::{self, Cursor, Read, Write};
+use std;
+use super::num::ToPrimitive;
+use self::byteorder::{BigEndian, ReadBytesExt, WriteBytesExt};
+
+#[path = "tests.rs"]
+#[cfg(test)]
+mod tests;
+
+#[path = "benchmarks.rs"]
+#[cfg(all(test, feature = "bench_private"))]
+mod benchmarks;
+
+const V2_COOKIE_BASE: u32 = 0x1c849303;
+
+const V2_COOKIE: u32 = V2_COOKIE_BASE | 0x10;
+
+const V2_HEADER_SIZE: usize = 40;
+
+/// Errors that occur during serialization.
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub enum SerializeError {
+    /// A count above i64::max_value() cannot be zig-zag encoded, and therefore cannot be
+    /// serialized.
+    CountNotSerializable,
+    /// Internal calculations cannot be represented in `usize`. Use smaller histograms or beefier
+    /// hardware.
+    UsizeTypeTooSmall,
+    /// An i/o operation failed.
+    IoError
+}
+
+impl std::convert::From<std::io::Error> for SerializeError {
+    fn from(_: std::io::Error) -> Self {
+        SerializeError::IoError
+    }
+}
+
+/// Serializer for the V2 binary format.
+/// Serializers are intended to be re-used for many histograms.
```

Comment:
What is the reasoning behind having a separate `struct` for this, rather than just having methods on `Histogram`? You could still keep the implementation in a separate file by adding a trait (i.e., `Serialize`), and then implementing it for `Histogram`.